### PR TITLE
Second try at scenarization

### PR DESCRIPTION
**Pull request by flomnes**
Following @pet-mit's remarks

### CODE REVIEW

File: src/solver/optim/api/include/antares/optim/api/LinearProblemData.h [hunk 1]
```diff
@@ -27,45 +27,78 @@
...
-    class LinearProblemData final
+class LinearProblemData final
+{
+public:
+    using ScalarDataDict = std::map<std::string, std::vector<double>>;
+
+    using TimedData = std::vector<double>;
+    using TimedDataDict = std::map<std::string, std::vector<TimedData>>;
+
+    double getScalarData(const std::string& key, int scenario = 0) const
```
Comment (flomnes) [inline src/solver/optim/api/include/antares/optim/api/LinearProblemData.h]: Use a default value for compatibility, to be removed later

File: src/solver/optim/api/include/antares/optim/api/LinearProblemData.h [hunk 2]
```diff
@@ -27,45 +27,74 @@
...
+    using TimedDataDict = std::map<std::string, std::vector<TimedData>>;
+
+private:
+    // TODO : timestamps or timesteps?
+    std::vector<int> timeStamps_;
+    int timeResolutionInMinutes_;
+
+    ScalarDataDict scalarData_;
+    TimedDataDict timedData_;
+    // TODO : handle scenarios, and data vectorized on scenarios, on time, or on both
```
Comment (pet-mit) [inline src/solver/optim/api/include/antares/optim/api/LinearProblemData.h:51]: remove this TODO

File: src/solver/optim/api/include/antares/optim/api/LinearProblemData.h [hunk 3]
```diff
@@ -27,45 +27,74 @@
...
 namespace Antares::optim::api
 {
-    class LinearProblemData final
+class LinearProblemData final
+{
+public:
+    using ScalarDataDict = std::map<std::string, std::vector<double>>;
+
+    using TimedData = std::vector<double>;
+    using TimedDataDict = std::map<std::string, std::vector<TimedData>>;
```
Comment (pet-mit) [inline src/solver/optim/api/include/antares/optim/api/LinearProblemData.h:42]: If I understand well, this implicitely means that 1st dimension is scenario and 2nd dimension is time.
I find it hard to read & maintain in time and over different files. Is there a way to have a multi-key map instead? Or define an extra struct / class? Or at least explicit it in the name of the attribute?

Comment (flomnes) [reply src/solver/optim/api/include/antares/optim/api/LinearProblemData.h:42]: Sure !

**Pull request closed without merging**